fs/src/io_uring/memory.rs

```diff
@@ -14,64 +15,60 @@ use {
 // and chunk it in slices of up to 1G each.
 const FIXED_BUFFER_LEN: usize = 1024 * 1024 * 1024;
 
-pub enum LargeBuffer {
-    Vec(Vec<u8>),
-    HugeTable(PageAlignedMemory),
-}
-
-impl Deref for LargeBuffer {
-    type Target = [u8];
+#[derive(Debug)]
+struct AllocError;
 
-    fn deref(&self) -> &Self::Target {
-        match self {
-            Self::Vec(buf) => buf.as_slice(),
-            Self::HugeTable(mem) => mem.deref(),
-        }
-    }
+enum AllocationMethod {
+    Std,
+    Mmap,
 }
 
-impl DerefMut for LargeBuffer {
-    fn deref_mut(&mut self) -> &mut Self::Target {
-        match self {
-            Self::Vec(buf) => buf.as_mut_slice(),
-            Self::HugeTable(mem) => mem.deref_mut(),
-        }
-    }
-}
-
-impl AsMut<[u8]> for LargeBuffer {
-    fn as_mut(&mut self) -> &mut [u8] {
-        match self {
-            Self::Vec(vec) => vec.as_mut_slice(),
-            LargeBuffer::HugeTable(mem) => mem,
-        }
-    }
+/// `PageAlignedMemory.ptr` is always aligned to the page size.
```

Comment:
Doc comment should have a one-line summary at the top, newline and then expanded description block

Reply:
https://github.com/anza-xyz/agave/pull/9531/commits/db2b84893f162a1904a07f3c7fdc45a8ca12d231